File: integration_examples/example.live.pl
#!/usr/local/cpanel/3rdparty/bin/perl
# cpanel - base/frontend/paper_lantern/test.live.pl
#                                                    Copyright 2014 cPanel, Inc.
#                                                           All rights Reserved.
# copyright@cpanel.net                                         http://cpanel.net
# This code is subject to the cPanel license. Unauthorized copying is prohibited

use strict;

use Cpanel::LiveAPI ();

my $cpanel = Cpanel::LiveAPI->new();

print "Content-type: text/html\r\n\r\n";

print $cpanel->header('Example perl Page');

print "We recommend that you create a new template toolkit file and place it in the paper lantern theme's root directory.";

print $cpanel->footer();

$cpanel->end();

File: integration_examples/example.live.php
<?php
/* This is an example of how to use LiveAPI with paper_lantern
 * paper_lantern has adopted a bootstrap CSS base, for information on CSS classes to use
 * please see: http://bootstrap.com
 *
 * As for method available to the $cpanel object in the example, cpanel.php contains
 * etensive inline documentation in the PHP Doc format.  Taking a look at it is highly advised.
 *
 * You can also look at the test.live.php file shipped along side this example.
 */

include("/usr/local/cpanel/php/cpanel.php");
$cpanel = new CPANEL();
print $cpanel->header( "Some Example Page" );
?>

<div class="container">
	<div class="row">
		Here's a list of environment variables!<br />
	</div>
	<div class="row">
		<table class="table">
			<thead>
				<tr>
					<td>
						Name
					</td>
					<td>
						Something
					</td>
				</tr>
			</thead>
			<tbody>
<?php
foreach ($_ENV as $k => $v) {
	echo "<tr><td>$k</td><td>$v</td></tr>";
}
?>
			</tbody>
		</table>
	</div>
	<div>
		<button type="button" class="btn btn-primary active">Do nothing!</button><br />
	</div>
</div>
<?php
print $cpanel->footer();
$cpanel->end();
?>
